### ISSUE TITLE
0009430:
Make tine work with davdroid

### ISSUE THREAD

**Issue by Gloirin** [tine20/Tine-2.0-Open-Source-Groupware-and-CRM#4734]
**Reported by mm on 21 Dec 2013 21:09**

**Version:** Collin (2013.10.3)

Calendar sync works, but update of entries is not possible.
https://github.com/rfc2822/davdroid

**Steps to reproduce:** Try to update any calendar entry from the mobile phone.

**Additional information:** 0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setupLogger::577 Logger initialized
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::createAndConfigureDbAdapter::956 Creating Pdo_Mysql DB adapter
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setupCache::668 cache disabled
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setupLogger::577 Logger initialized
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setupCache::668 cache disabled
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setSessionBackend::904 Session of backend type &#039;File&#039; configured.
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setupUserLocale::1167 Setting user locale: en
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Server_WebDAV::handle::46 is CalDav, CardDAV or WebDAV request.
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setExecutionLifeTime::1244 setting execution life time to: 0
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Auth::authenticate::213 Authentication of mm succeeded
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_User::setBackendType::272 Setting backend type to Sql
0adc7 -- none -- - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_User::factory::233 Created user backend of type Sql
0adc7 mm - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Core::setupUserLocale::1167 Setting user locale: en
0adc7 mm - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Server_WebDAV::handle::72 requestUri:/calendars/2/40/20131221T205909Z-13567_7f8e495b6a94df1b.ics
0adc7 mm - 2013-12-21T21:02:50+00:00 WARN (4): Calendar_Frontend_WebDAV_Event::put::351 update by generic client not allowed. See Calendar_Convert_Event_VCalendar_Factory for supported clients.
0adc7 mm - 2013-12-21T21:02:50+00:00 INFO (6): Tinebase_Controller_Record_Abstract::update::747 Update Tinebase_Model_AccessLog
0adc7 mm - 2013-12-21T21:02:50+00:00 INFO (6): index.php (25) METHOD: Tinebase_Server_WebDAV::PUT / TIME: 0.47972297668457 seconds /  Memory usage: 8.5 MB /  Real patch cache size: 16372

**Comment by Gloirin** [comment]
**Comment posted by mm on 21 Dec 2013 21:42**

I just removed the check for genreric client and it works for me. I konw that I&#039;m opening a can of worms, but I&#039;m the only one using that instance of tine.
But I guess the only thing needed is to add davdroid to the known clients as tine uses sabredav and davdroid is known to work with other sabredav servers like owncloud.

--- ./Calendar/Frontend/WebDAV/Event.php.bak    2013-12-21 22:38:21.815442879 +0100
+++ ./Calendar/Frontend/WebDAV/Event.php        2013-12-21 22:18:53.129781800 +0100
@@ -346,11 +346,11 @@
     public function put($cardData) 
     {
         $this-&gt;_assertEventFilter();
-        if (get_class($this-&gt;_converter) == &#039;Calendar_Convert_Event_VCalendar_Generic&#039;) {
-            if (Tinebase_Core::isLogLevel(Zend_Log::WARN)) 
-                Tinebase_Core::getLogger()-&gt;warn(__METHOD__ . &#039;::&#039; . __LINE__ . &quot; update by generic client not allowed. See Calendar_Convert_Event_VCalendar_Factory for supported clients.&quot;);
-            throw new Sabre\DAV\Exception\Forbidden(&#039;Update denied for unknown client&#039;);
-        }
+        //if (get_class($this-&gt;_converter) == &#039;Calendar_Convert_Event_VCalendar_Generic&#039;) {
+        //   if (Tinebase_Core::isLogLevel(Zend_Log::WARN)) 
+        //        Tinebase_Core::getLogger()-&gt;warn(__METHOD__ . &#039;::&#039; . __LINE__ . &quot; update by generic client not allowed. See Calendar_Convert_Event_VCalendar_Factory for supported clients.&quot;);
+        //    throw new Sabre\DAV\Exception\Forbidden(&#039;Update denied for unknown client&#039;);
+        //}
         
         if (is_resource($cardData)) {
             $cardData = stream_get_contents($cardData);

**Comment by Gloirin** [comment]
**Comment posted by ingoratsdorf on 12 Dec 2014 02:13**

We could just add a new converter for DAVdroid. Just need to figure out what it is sending and how and what the user agent string is.
Simple.
Now that I have switched from iPhone to Android, I might give this a try.

**Comment by Gloirin** [comment]
**Comment posted by ingoratsdorf on 12 Dec 2014 05:05**

Okay, step one: user agent:
&quot;PROPFIND /calendars/c21349a5775f4dd6e603214dc0ad83ffc37729b6/29/ HTTP/1.1&quot; 207 1167 &quot;-&quot; &quot;DAVdroid/0.6.8&quot;

I am onto it

**Comment by Gloirin** [comment]
**Comment posted by ingoratsdorf on 12 Dec 2014 05:06**

I&#039;ll be building a new converter for this. Will take a few weeks of testing.

**Comment by Gloirin** [comment]
**Comment posted by ingoratsdorf on 17 Dec 2014 18:30**

I have tried DAVdroid and it has some issues:

a) It does not accept self-signed certificates, that most SME&#039;s and private people would use. Only with workarounds that have profound effect on your device safety. You can install your self-signed cert, but then this will restrict your lock screen options to PIN or passcode only.

b) All events are offset by an hour. Maybe this is a general issue, probably rather once again only me living in New Zealand with GMT+13. Anyway, annoying.

So in summary, I won&#039;t provide a cobverter for it.
I have uninstalled DAVdroid from my device.

Nga mihi,
Ingo
